drone-webdriver/src/test/java/org/jboss/arquillian/drone/webdriver/factory/ChromeHeadlessDriverTest.java

```diff
@@ -0,0 +1,69 @@
+package org.jboss.arquillian.drone.webdriver.factory;
+
+import java.io.IOException;
+import java.net.URL;
+import org.assertj.core.api.Assertions;
+import org.jboss.arquillian.drone.webdriver.configuration.WebDriverConfiguration;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.chrome.ChromeOptions;
+import org.openqa.selenium.remote.DesiredCapabilities;
+
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ChromeHeadlessDriverTest {
+
+    @BeforeClass
+    public static void executeOnlyIfChromeHeadlessBrowser() {
+        String browser = System.getProperty("browser");
+        Assume.assumeTrue(browser.equals("chromeHeadless") || browser.equals("chromeheadless"));
+    }
+
+    @Test
+    public void testOpenSimplePageUsingHeadlessChrome() throws IOException {
+        ChromeDriverFactory chromeDriverFactory = new ChromeDriverFactory();
+
+        DesiredCapabilities chromeCaps =
+            new DesiredCapabilities(new BrowserCapabilitiesList.chromeHeadless().getRawCapabilities());
+        WebDriverConfiguration configuration = getMockedConfiguration(chromeCaps);
+
+        WebDriver driver = chromeDriverFactory.createInstance(configuration);
+        URL page = this.getClass().getClassLoader().getResource("simple.html");
+        driver.get(page.toString());
+        Assert.assertEquals("The page title doesn't match.", "Simple Page", driver.getTitle());
+        driver.quit();
+    }
+
+    @Test
+    public void shouldSetChromeOptionsForHeadlessChromeHeadlessDriver() throws IOException {
```

Comment:
This test doesn't require the browser set - it doesn't require the browser installed on PC. In other words, you can run it even when the `-Dbrowser=chromeHeadless` is not set.